clang/lib/Format/TokenAnnotator.cpp

```diff
@@ -4589,9 +4589,12 @@ bool TokenAnnotator::spaceRequiredBetween(const AnnotatedLine &Line,
     if (!BeforeLeft)
       return false;
     if (BeforeLeft->is(tok::coloncolon)) {
-      const auto *Prev = BeforeLeft->Previous;
-      return Left.is(tok::star) && Prev &&
-             !Prev->endsSequence(tok::identifier, TT_FunctionTypeLParen);
+      if (Left.isNot(tok::star))
+        return false;
+      if (!Right.startsSequence(tok::identifier, tok::r_paren))
+        return true;
+      const auto *Tok = Right.Next->MatchingParen;
```

Comment:
`Right.Next` might be nullptr

Reply:
No, because `startsSequence()` returned `true`. I'll add an assertion though.